proposals/0759-react-native-frameworks.md

```diff
@@ -0,0 +1,452 @@
+- **Set “the standard” for the ecosystem** - React Native should set standards that are general enough to work across Frameworks so that Frameworks can use our standards as a foundation to build tools and extensions on top of them.
+- **Enable portability** - React Native should provide all of the standards and language features needed to write libraries that are portable to use across all frameworks (either bespoke or Open Source Frameworks).
+- **Fostering a healthy community** - As mentioned, by defining our boundaries, it will be clearer where we draw the line between React Native Core APIs and third-party framework APIs, we’ll save a lot of confusion to our users, and we’ll avoid re-implementing solutions.
+
+## Frameworks & Platforms
+
+Before we delve into the responsibilities of React Native & Frameworks, it’s crucial to define what is a **Platform**, as one of the key value propositions is to let you write applications and target several different Platforms.
+
+A platform is the underlying computing architecture and operating system on which applications run.
+
+React Native offers **first-party support** for Android & iOS, as the code to target Android & iOS lives inside the [react-native repository](https://github.com/facebook/react-native). Such code is developed and maintained by Meta.
+
+Over the years, the ecosystem has developed support for other platforms which we call [Out-of-Tree platforms (OOT)](https://reactnative.dev/docs/out-of-tree-platforms).
+
+Each OOT platform comes with its own documentation, setup instructions, and tools/libraries to develop applications on the new target platform.
+
+We consider the Framework and Platform concepts as **orthogonal**, i.e. you can use a Framework to ship to multiple platforms, and a Platform can be targeted by multiple Frameworks. In this context the role of a Framework is to abstract complexity and make it easier to ship to multiple Platforms providing a cohesive Developer Experience (see below).
+
+## Responsibilities
+
+This section describes where we draw the line between the React Native package and Frameworks.
+
+> [!NOTE]
+> For the sake of simplicity, we will be focusing on the React Native package (i.e. the NPM package called `react-native` which offers first-party support for Android & iOS).
+
+### What does the React Native package do
+
+The React Native package is responsible for **core APIs and capabilities** to build native apps. React Native is also responsible for offering **APIs** that allow Frameworks to deliver their experience. When such APIs are missing, it’s our interest to create them and have a conversation with Framework developers to inform their development, and help framework developers merge changes back into React Native.
```

Comment:
i believe the idea is that in a future RFC, or series of RFCs, the plan is to go into more detail on what the exact core APIs and capabilities will be. that may be worth calling out here, so folks know that this information is coming

Reply:
Yup added in 2387901